django/views/generic/base.py

```diff
@@ -30,6 +32,27 @@ def get_context_data(self, **kwargs):
         return kwargs
 
 
+class ViewType(Enum):
+    SYNC = auto()
+    ASYNC = auto()
+
+    @classmethod
+    def of_view(cls, view):
+        handlers = [
+            getattr(view, method)
+            for method in view.http_method_names
+            if (method != "options" and hasattr(view, method))
+        ]
+        if not handlers:
+            return cls.SYNC
+        is_async = asyncio.iscoroutinefunction(handlers[0])
+        if not all(asyncio.iscoroutinefunction(h) == is_async for h in handlers[1:]):
+            raise ImproperlyConfigured(
+                "View subclasses HTTP handlers must either be all sync or all async."
```

Comment:
I'd add a view name to the exception message, e.g.: 
```suggestion
                f"{view.__qualname__} HTTP handlers must either be all sync or all async."
```